File: JobApplicationMgtSystem/JobApplicationApi/JobApplicationApi/Models/InterviewStage.cs
namespace JobApplicationApi.Models
{
    public class InterviewStage
    {
        public int Id { get; set; }
        public string StageName { get; set; } = string.Empty;  // "Phone screen", "Technical", "Final"
        public InterviewType Type { get; set; }
        public DateOnly? ScheduledAt { get; set; }
        public string? Location { get; set; }
        public string? Interviewer { get; set; }
        public InterviewStatus Status { get; set; } = InterviewStatus.Scheduled;
        public string? Notes { get; set; }

        public int JobApplicationId { get; set; }
        public JobApplication JobApplication { get; set; } = null!;
    }

    public enum InterviewType
    {
        Phone,
        Video,
        OnSite,
        Technical,
        Behavioral,
        Final
    }

    public enum InterviewStatus
    {
        Scheduled,
        Completed,
        Cancelled,
        Rescheduled,
        Passed,
        Failed
    }
}

File: JobApplicationMgtSystem/JobApplicationApi/Frontend/jobapp-frontend/src/types/interviewStage.ts
export interface InterviewStage 
{
  id: number;
  stageName: string;
  type: InterviewType;
  location: string;
  interviewer: string;
  status: InterviewStatus;
  scheduledAt: string; // DateOnly → "YYYY-MM-DD"
  notes: string;
  isComplete: boolean;
}

export const InterviewType =
{
  Phone: 0,
  Video: 1,
  OnSite: 2,
  Technical: 3,
  Behavioral: 4,
  Final: 5,
} as const;

export type InterviewType = typeof InterviewType[keyof typeof InterviewType];

export const interviewTypeLabels: Record<InterviewType, string> = 
{
    [InterviewType.Phone]: "Phone",
    [InterviewType.Video]: "Video",
    [InterviewType.OnSite]: "On-Site",
    [InterviewType.Technical]: "Technical",
    [InterviewType.Behavioral]: "Behavioral",
    [InterviewType.Final]: "Final",
}



export const InterviewStatus = 
{
  Scheduled: 0,
  Completed: 1,
  Cancelled: 2,
  Rescheduled: 3,
  Passed: 4,
  Failed: 5,
} as const;

export type InterviewStatus = typeof InterviewStatus[keyof typeof InterviewStatus];

export const interviewStatusLabels: Record<InterviewStatus, string> = 
{
    [InterviewStatus.Scheduled]: "Scheduled",
    [InterviewStatus.Completed]: "Completed",
    [InterviewStatus.Cancelled]: "Cancelled",
    [InterviewStatus.Rescheduled]: "Rescheduled",
    [InterviewStatus.Passed]: "Passed",
    [InterviewStatus.Failed]: "Failed",
}